src/internal/_curryN.js

```diff
@@ -0,0 +1,38 @@
+var arity = require('../arity');
+
+
+/**
+ * Internal curryN function.
+ *
+ * @private
+ * @category Function
+ * @param {Number} length The arity of the curried function.
+ * @return {array} An array of arguments received thus far.
+ * @param {Function} fn The function to curry.
+ */
+module.exports = function _curryN(length, recieved, fn) {
+  return function() {
+    var combined = [];
+    var argsIdx = 0;
+    var left = length;
+    var combinedIdx = 0;
+    while (combinedIdx < recieved.length || argsIdx < arguments.length) {
+      var result;
+      var rec = recieved[combinedIdx];
+      var recNotPlaceholder = rec == null || rec['@@functional/placeholder'] !== true;
```

Comment:
I'd also like to point out that the helper could easily test explicitly for `undefined` and `null` instead of relying on type coercion like we currently do. I've experienced performance gains from that in the past.